pymeasure/instruments/aimtti/ld400p.py

```diff
@@ -0,0 +1,126 @@
+    """
+
+    unit_suffixes = ["A", "V", "W", "OHM", "SIE", "HZ"]
+
+    @classmethod
+    def remove_unit_suffix(cls, msg: str) -> str:
+        """Remove known unit suffixes from a msg.
+
+        The units of some replies depend on the mode of the device, hence all known
+        units are removed.
+        """
+        for suffix in cls.unit_suffixes:
+            msg = msg.replace(suffix, "")
+        return msg
+
+    def __init__(self, adapter, name="LD400", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    mode = Instrument.control(
+        "MODE?",
+        "MODE %s",
+        """Control the mode of the digital load. Can be one of:
+        ``("C", "P", "R", "G", "V")``, corresponding to (respectively) constant current,
+        power, resistance, conductance or voltage (string).
+        """,
+        validator=strict_discrete_set,
+        values=["C", "P", "R", "G", "V"],
+        get_process=lambda v: v.replace("MODE ", ""),
+    )
+
+    level_select = Instrument.control(
+        "LVLSEL?",
+        "LVLSEL %s",
+        """Control the selected level of the digital load. Can be one of:
+        ``("A", "B", "T", "V", "E")``, corresponding to (respectively) Level A, Level B,
+        Transient, Ext Voltage and Ext TTL (string).
+        """,
+        validator=strict_discrete_set,
+        values=["A", "B", "T", "V", "E"],
+        get_process=lambda v: v.replace("LVLSEL ", ""),
+    )
+
+    level_a = Instrument.control(
+        "A?",
+        "A %g",
+        """Control the A level of the current control mode (float).""",
+        preprocess_reply=lambda v: LD400P.remove_unit_suffix(v.replace("A", "")),
+    )
+
+    level_b = Instrument.control(
+        "B?",
+        "B %g",
+        """Control the B level of the current control mode (float).""",
+        preprocess_reply=lambda v: LD400P.remove_unit_suffix(v.replace("B", "")),
+    )
+
+    input_enabled = Instrument.control(
+        "INP?",
+        "INP %d",
+        """Control the input state of the digital load, can be ``True`` (on) or
+        ``False`` (off).
+        """,
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+        preprocess_reply=lambda v: v.replace("INP ", ""),
+    )
+
+    # Voltage / current replies are _not_ prefixed by the command, but do have a unit:
+
+    voltage = Instrument.measurement(
+        "V?",
+        """Get the measured source input voltage (float).""",
+        preprocess_reply=lambda v: v.replace("V", ""),
+    )
+
+    current = Instrument.measurement(
+        "I?",
+        """Get the measured load current (float).""",
+        preprocess_reply=lambda v: v.replace("A", ""),
+    )
+
+    # Remove SCP properties that don't work on this device:
+    options = None
+    next_error = None
+
+    def check_errors(self):
+        raise NotImplementedError("This method is not available on this device")
```

Comment:
@BenediktBurger I choose to un-define properties that don't work from `SCPIMixIn`, is that okay?
The alternative is to define the properties that do work, but to me that's less DRY.

Reply:
For some reason code like
```python
    def __init__(self, adapter, name="LD400", **kwargs):
        super().__init__(adapter, name, **kwargs)

        del self.options
```
keeps throwing the error `AttributeError: 'LD400P' object has no attribute 'options'`. Was this the code you had in mind?

Reply:
Yes, that what, was I had in mind.

Reply:
I just released my comment was ambiguous: the above doesn't work because an `AttributeError` is raised during _deletion_, not afterwards. So this snippet doesn't work. Do you have an example where it's used before?

Reply:
I just did tests, confirming your findings.
So, yes, your raising "NotImplementedError" is fine.

Reply:
I went with another option, to bring up a real `AttributeError`:
```python
    @property
    def options(self):
        raise AttributeError("No attribute 'options' exists")
```

Reply:
You could add a suitable docstring (it has to start with set, get, measure, or control).
Here for example: "Get the options - does not exist for this device".

Reply:
No, the errors are more complicated, I didn't see a quick way to fix it better: https://github.com/pymeasure/pymeasure/actions/runs/11418311223/job/31771599574#step:7:239

Reply:
It tries to split the docstring. If you do not have a docstring (llike in your case), it is None, which is unsplittable.
Adding a docstring should fix it.